torchtext/prototype/generate.py

```diff
@@ -75,10 +88,21 @@ def greedy_search(
             outputs = self.model(**model_inputs)
             output_key = "logits" if self.is_huggingface_model else "decoder_output"
             decoder_output = outputs[output_key]
-
-            # Calculate probabilities and take the most likely next token
-            probs = F.log_softmax(decoder_output[:, -1], dim=-1)
-            next_tokens = torch.argmax(probs, dim=-1)
+            log_probs = F.log_softmax(decoder_output[:, -1], dim=-1)
+
+            if do_sample:
+                probs = log_probs.softmax(dim=-1)
```

Comment:
Taking double softmax here? Probs are already softmax'd on L91

Reply:
Thanks for spotting this ! It should be a `.exp` to retrieve the original probabilities indeed.